From 5b0624df4629a87c759b4dade9687adbd55fafd9 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 01 Jan 2023 23:27:48 +0200
Subject: [PATCH] [SW-115685] common: fit: read tpm devid

By demand, printout the TPM devid, on the FW console.

A FIT with new load type - FIT_LOAD_TYPE_READ_TPM_DEVID,
is created. The FW will print out the TPM devid.
It is a complementary FIT to the existing
FIT_LOAD_TYPE_WRITE_TPM_DEVID

Change-Id: If37e54a36ebac8dbe777b1cde3a2748f7ad1593a
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/make/gaudi2/Makefile b/scripts/make/gaudi2/Makefile
index 5b0c18f..a0378a5 100644
--- a/scripts/make/gaudi2/Makefile
+++ b/scripts/make/gaudi2/Makefile
@@ -84,6 +84,14 @@
 gaudi2-write-tpm-devid-install:
 	-sshpass -p$(HL_PASSWD) $(SCP) $(IMAGES_DIR)/${HL_DEV}-write_tpm_devid.itb root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 
+.PHONY: gaudi2-read-tpm-devid-create
+gaudi2-read-tpm-devid-create:
+	${BR2_EXTERNAL}/board/habanalabs/common/create-simple.sh read_tpm_devid fw-loader-conf no-use-elbi no-use-fwupdate load-bootloader
+
+.PHONY: gaudi2-read-tpm-devid-install
+gaudi2-read-tpm-devid-install:
+	-sshpass -p$(HL_PASSWD) $(SCP) $(IMAGES_DIR)/${HL_DEV}-read_tpm_devid.itb root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
+
 .PHONY: gaudi2-otp-local-create
 gaudi2-otp-local-create:
 		$(HOST_MAKE_ENV) ${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}/create-otp-local-image.sh
@@ -149,6 +157,7 @@
 	${HL_DEV}-nic-dis.itb		\
 	${HL_DEV}-eeprom.itb		\
 	${HL_DEV}-write_tpm_devid.itb		\
+	${HL_DEV}-read_tpm_devid.itb		\
 	${HL_DEV}-slt_set.itb			\
 	${HL_DEV}-security_enable.itb		\
 	${HL_DEV}-otp.itb				\
